### PR TITLE
Reduce system token precision

### PR DESCRIPTION
**Pull request by James-Mart**
* Sets max precision to 8 instead of 16
* Updates the precision class so min/max are enforced when unpacking from frackpack or json
* Updates the precision of the system token to 4
* Updates a bunch of tests to look up the previously hardcoded sys token precision

### CODE REVIEW

File: services/user/Tokens/include/services/user/tokenTypes.hpp [hunk 1]
```diff
@@ -12,26 +12,55 @@ namespace UserService
...
+                           std::to_string(precisionMin) + " to " + std::to_string(precisionMax));
+         }
+         return true;
       }
 
       friend std::strong_ordering operator<=>(const Precision&, const Precision&) = default;
    };
    PSIO_REFLECT(Precision, value);
 
+   void from_json(Precision& p, auto& stream)
```
Comment (James-Mart) [inline services/user/Tokens/include/services/user/tokenTypes.hpp:38]: This is essentially a simplified version of the struct unpacking [here](https://github.com/gofractally/psibase/blob/main/libraries/psio/include/psio/from_json.hpp#L745).

File: services/user/Tokens/include/services/user/tokenTypes.hpp [hunk 2]
```diff
@@ -12,26 +12,55 @@ namespace UserService
...
+                                                           from_json(p.*member, stream);
+                                                           found = true;
+                                                        });
+                          if (not found)
+                          {
+                             from_json_skip_value(stream);
+                          }
+                       });
+
+      Precision::validate(p.value);
```
Comment (swatanabe) [inline services/user/Tokens/include/services/user/tokenTypes.hpp:56]: I think we'd be better off making `from_json` call `clio_validate_packable`, to get unified validation for deserialization.

Comment (James-Mart) [reply services/user/Tokens/include/services/user/tokenTypes.hpp]: done

Comment (swatanabe) [reply services/user/Tokens/include/services/user/tokenTypes.hpp]: Not quite what I meant. I meant making the generic from_json call `clio_validate_packable` if the call is well-formed.

File: services/user/Tokens/include/services/user/tokenTypes.hpp [hunk 3]
```diff
@@ -12,26 +12,34 @@ namespace UserService
 
    struct Precision
    {
-      static constexpr uint8_t          precisionMin = 0;
-      static constexpr uint8_t          precisionMax = 16;
-      static constexpr std::string_view errorInvalid = "Value exceeds allowed Precision range";
+      static constexpr uint8_t precisionMin = 0;
+      static constexpr uint8_t precisionMax = 8;
```
Comment (swatanabe) [inline services/user/Tokens/include/services/user/tokenTypes.hpp:16]: I don't understand the reasoning for choosing a particular value of `precisionMax`. Is there some way that 16 causes a problem?

**Review by swatanabe — APPROVED** [review]

File: services/user/Tokens/include/services/user/tokenTypes.hpp (hunk 3 above)
Comment (James-Mart) [reply services/user/Tokens/include/services/user/tokenTypes.hpp:16]: It's just for ergonomics. Can be increased later if there's actually a need for >8 decimals of precision.

**Pull request merged**